### ISSUE TITLE
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: OccupiedError 

### ISSUE THREAD

**Issue by kwannoel** [risingwavelabs/risingwave#10493]
### Describe the bug

Occupied error thrown when processing updates to join state

I'm able to reproduce by running it locally.

### Error message/log

```text
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: OccupiedError { key: [0, 128, 0, 0, 0, 0, 0, 2, 103, 0, 128, 0, 0, 0, 98, 97, 99, 96, 0, 0, 0, 0], old_value: EncodedJoinRow { compacted_row: CompactedRow { row: b"\x01\n\0\0\0wjd1SVkpiL\x01V\xc3ab\0\0\0\0\0\0\0\0\x01\xd5\x14cb\0\0\0\0\0\0\0\0\x01\0\0\0\0\0\0\0\x80\x01\n\0\0\0z0NQ592cok\x01]\x02\0\0\0\0\0\0\x01g\x02\0\0\0\0\0\0\x01`cab\0\0\0\0\0\0\0\0" }, degree: 0 }, new_value: EncodedJoinRow { compacted_row: CompactedRow { row: b"\x01\n\0\0\0UmT3KaDusR\x01\xdd\x14cb\0\0\0\0\0\0\0\0\x01\xd5\x14cb\0\0\0\0\0\0\0\0\x01Z\x03\0\0\0\0\0\0\x01\n\0\0\0ddB77tctil\x01]\x02\0\0\0\0\0\0\x01g\x02\0\0\0\0\0\0\x01`cab\0\0\0\0\0\0\0\0" }, degree: 0 } }', src/stream/src/executor/managed_state/join/join_entry_state.rs:54:44
stack backtrace:
   0: rust_begin_unwind
             at /rustc/f0411ffcebcd7f75ac02ed45feb53ffd07b75398/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/f0411ffcebcd7f75ac02ed45feb53ffd07b75398/library/core/src/panicking.rs:67:14
   2: core::result::unwrap_failed
             at /rustc/f0411ffcebcd7f75ac02ed45feb53ffd07b75398/library/core/src/result.rs:1651:5
   3: unwrap<&mut risingwave_stream::executor::managed_state::join::EncodedJoinRow, alloc::collections::btree::map::entry::OccupiedError<alloc::vec::Vec<u8, alloc::alloc::Global>, risingwave_stream::executor::managed_state::join::EncodedJoinRow, alloc::alloc::Global>>
             at /rustc/f0411ffcebcd7f75ac02ed45feb53ffd07b75398/library/core/src/result.rs:1076:23
   4: insert
             at ./src/stream/src/executor/managed_state/join/join_entry_state.rs:54:9
   5: {async_fn#0}<risingwave_common::hash::key_v2::HashKeyImpl<risingwave_common::hash::key_v2::StackStorage<16>, risingwave_common::hash::key::StackNullBitmap>, risingwave_storage::monitor::monitored_store::MonitoredStateStore<alloc::boxed::Box<dyn risingwave_storage::store_impl::boxed_state_store::DynamicDispatchedStateStore, alloc::alloc::Global>>, risingwave_common::array::data_chunk_iter::RowRef>
             at ./src/stream/src/executor/managed_state/join/mod.rs:507:13
   6: {generator#0}<risingwave_common::hash::key_v2::HashKeyImpl<risingwave_common::hash::key_v2::StackStorage<16>, risingwave_common::hash::key::StackNullBitmap>, risingwave_storage::monitor::monitored_store::MonitoredStateStore<alloc::boxed::Box<dyn risingwave_storage::store_impl::boxed_state_store::DynamicDispatchedStateStore, alloc::alloc::Global>>, 1, 1>
             at ./src/stream/src/executor/hash_join.rs:1126:65
   7: poll_next<risingwave_stream::executor::hash_join::{impl#6}::eq_join_oneside::{generator_env#0}<risingwave_common::hash::key_v2::HashKeyImpl<risingwave_common::hash::key_v2::StackStorage<16>, risingwave_common::hash::key::StackNullBitmap>, risingwave_storage::monitor::monitored_store::MonitoredStateStore<alloc::boxed::Box<dyn risingwave_storage::store_impl::boxed_state_store::DynamicDispatchedStateStore, alloc::alloc::Global>>, 1, 1>, risingwave_common::array::stream_chunk::StreamChunk, risingwave_stream::executor::error::StreamExecutorError>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.6/src/lib.rs:463:33
   8: {generator#0}<risingwave_common::hash::key_v2::HashKeyImpl<risingwave_common::hash::key_v2::StackStorage<16>, risingwave_common::hash::key::StackNullBitmap>, risingwave_storage::monitor::monitored_store::MonitoredStateStore<alloc::boxed::Box<dyn risingwave_storage::store_impl::boxed_state_store::DynamicDispatchedStateStore, alloc::alloc::Global>>, 1>
             at ./src/stream/src/executor/hash_join.rs:692:5
   9: poll_next<risingwave_stream::executor::hash_join::{impl#6}::into_stream::{generator_env#0}<risingwave_common::hash::key_v2::HashKeyImpl<risingwave_common::hash::key_v2::StackStorage<16>, risingwave_common::hash::key::StackNullBitmap>, risingwave_storage::monitor::monitored_store::MonitoredStateStore<alloc::boxed::Box<dyn risingwave_storage::store_impl::boxed_state_store::DynamicDispatchedStateStore, alloc::alloc::Global>>, 1>, risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.6/src/lib.rs:463:33
  10: poll<futures_util::stream::stream::next::Next<core::pin::Pin<&mut core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>, false>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/await-tree-0.1.1/src/future.rs:124:23
  11: {generator#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>
             at ./src/stream/src/executor/wrapper/trace.rs:126:10
  12: poll_next<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>, risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.6/src/lib.rs:463:33
  13: {generator#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>
             at ./src/stream/src/executor/wrapper/schema_check.rs:24:1
  14: poll_next<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>, risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.6/src/lib.rs:463:33
  15: {generator#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>
             at ./src/stream/src/executor/wrapper/epoch_check.rs:31:44
  16: poll_next<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>, risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.6/src/lib.rs:463:33
  17: try_poll_next_unpin<core::pin::Pin<&mut futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.28/src/stream/try_stream/mod.rs:1068:9
  18: poll<core::pin::Pin<&mut futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.28/src/stream/try_stream/try_next.rs:32:21
  19: {closure#0}<risingwave_common::util::epoch::EpochPair, futures_util::stream::try_stream::try_next::TryNext<core::pin::Pin<&mut futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/task/task_local.rs:348:35
  20: scope_inner<risingwave_common::util::epoch::EpochPair, tokio::task::task_local::{impl#3}::poll::{closure_env#0}<risingwave_common::util::epoch::EpochPair, futures_util::stream::try_stream::try_next::TryNext<core::pin::Pin<&mut futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>>, core::option::Option<core::task::poll::Poll<core::result::Result<core::option::Option<risingwave_stream::executor::Message>, risingwave_stream::executor::error::StreamExecutorError>>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/task/task_local.rs:233:19
  21: poll<risingwave_common::util::epoch::EpochPair, futures_util::stream::try_stream::try_next::TryNext<core::pin::Pin<&mut futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/task/task_local.rs:345:13
  22: {async_fn#0}<futures_util::stream::try_stream::try_next::TryNext<core::pin::Pin<&mut futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>>
             at ./src/common/src/util/epoch.rs:180:47
  23: {generator#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>
             at ./src/stream/src/executor/wrapper/epoch_provide.rs:31:59
  24: poll_next<risingwave_stream::executor::wrapper::epoch_provide::epoch_provide::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>, risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.6/src/lib.rs:463:33
  25: poll<futures_util::stream::stream::next::Next<core::pin::Pin<&mut futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_provide::epoch_provide::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-0.1.37/src/instrument.rs:272:9
  26: {generator#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_provide::epoch_provide::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>>
             at ./src/stream/src/executor/wrapper/trace.rs:46:69
  27: poll_next<risingwave_stream::executor::wrapper::trace::trace::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_provide::epoch_provide::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>>, risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.6/src/lib.rs:463:33
  28: {generator#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::trace::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_provide::epoch_provide::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>>>>
             at ./src/stream/src/executor/wrapper/update_check.rs:27:1
  29: poll_next<risingwave_stream::executor::wrapper::update_check::update_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::trace::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_provide::epoch_provide::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::epoch_check::epoch_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::schema_check::schema_check::{generator_env#0}<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::wrapper::trace::instrument_await_tree::{generator_env#0}<core::pin::Pin<alloc::boxed::Box<(dyn futures_core::stream::Stream<Item=core::result::Result<risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>> + core::marker::Send), alloc::alloc::Global>>>>>>>>>>>>>, risingwave_stream::executor::Message, risingwave_stream::executor::error::StreamExecutorError>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.6/src/lib.rs:463:33
  30: {generator#0}
             at ./src/stream/src/executor/dispatch.rs:258:9
  31: poll_next<risingwave_stream::executor::dispatch::{impl#2}::execute::{generator_env#0}, risingwave_stream::executor::Barrier, risingwave_stream::error::StreamError>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.6/src/lib.rs:463:33
  32: poll<risingwave_stream::executor::Barrier, risingwave_stream::error::StreamError, core::pin::Pin<alloc::boxed::Box<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::dispatch::{impl#2}::execute::{generator_env#0}>, alloc::alloc::Global>>>
             at /Users/noelkwan/.cargo/git/checkouts/tokio-968c02b7a1a41bea/ab251ad/tokio-stream/src/stream_ext/try_next.rs:43:9
  33: poll<tokio_stream::stream_ext::try_next::TryNext<core::pin::Pin<alloc::boxed::Box<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::dispatch::{impl#2}::execute::{generator_env#0}>, alloc::alloc::Global>>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-0.1.37/src/instrument.rs:272:9
  34: poll<tracing::instrument::Instrumented<tokio_stream::stream_ext::try_next::TryNext<core::pin::Pin<alloc::boxed::Box<futures_async_stream::try_stream::GenTryStream<risingwave_stream::executor::dispatch::{impl#2}::execute::{generator_env#0}>, alloc::alloc::Global>>>>, false>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/await-tree-0.1.1/src/future.rs:124:23
  35: {async_fn#0}<risingwave_stream::executor::dispatch::DispatchExecutor>
             at ./src/stream/src/executor/actor.rs:182:18
  36: poll<risingwave_stream::executor::actor::{impl#1}::run_consumer::{async_fn_env#0}<risingwave_stream::executor::dispatch::DispatchExecutor>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/future/maybe_done.rs:68:48
  37: {closure#0}<risingwave_stream::executor::dispatch::DispatchExecutor>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/macros/join.rs:126:24
  38: poll<(alloc::vec::Vec<(), alloc::alloc::Global>, core::result::Result<(), risingwave_stream::error::StreamError>), risingwave_stream::executor::actor::{impl#1}::run::{async_fn#0}::{closure_env#0}<risingwave_stream::executor::dispatch::DispatchExecutor>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/future/poll_fn.rs:58:9
  39: {async_fn#0}<risingwave_stream::executor::dispatch::DispatchExecutor>
             at ./src/stream/src/executor/actor.rs:154:9
  40: {async_block#2}
             at ./src/stream/src/task/stream_manager.rs:651:51
  41: {closure#0}<alloc::sync::Arc<lock_api::mutex::Mutex<parking_lot::raw_mutex::RawMutex, await_tree::context::TreeContext>>, risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/task/task_local.rs:348:35
  42: scope_inner<alloc::sync::Arc<lock_api::mutex::Mutex<parking_lot::raw_mutex::RawMutex, await_tree::context::TreeContext>>, tokio::task::task_local::{impl#3}::poll::{closure_env#0}<alloc::sync::Arc<lock_api::mutex::Mutex<parking_lot::raw_mutex::RawMutex, await_tree::context::TreeContext>>, risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>, core::option::Option<core::task::poll::Poll<()>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/task/task_local.rs:233:19
  43: poll<alloc::sync::Arc<lock_api::mutex::Mutex<parking_lot::raw_mutex::RawMutex, await_tree::context::TreeContext>>, risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/task/task_local.rs:345:13
  44: {async_fn#0}<risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/await-tree-0.1.1/src/registry.rs:51:45
  45: poll<await_tree::registry::{impl#1}::instrument::{async_fn_env#0}<risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>, risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.28/src/future/either.rs:109:32
  46: poll<futures_util::future::either::Either<await_tree::registry::{impl#1}::instrument::{async_fn_env#0}<risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>, risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-metrics-0.1.0/src/task.rs:2365:19
  47: {async_block#0}<tokio_metrics::task::Instrumented<futures_util::future::either::Either<await_tree::registry::{impl#1}::instrument::{async_fn_env#0}<risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>, risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.22/src/sim/task/mod.rs:629:28
  48: poll<madsim::sim::task::{impl#10}::spawn_inner::{closure#0}::{async_block_env#0}<tokio_metrics::task::Instrumented<futures_util::future::either::Either<await_tree::registry::{impl#1}::instrument::{async_fn_env#0}<risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>, risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>>>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.4.0/src/runnable.rs:455:26
  49: run<core::pin::Pin<alloc::boxed::Box<async_task::runnable::{impl#6}::spawn_local::Checked<madsim::sim::task::{impl#10}::spawn_inner::{closure#0}::{async_block_env#0}<tokio_metrics::task::Instrumented<futures_util::future::either::Either<await_tree::registry::{impl#1}::instrument::{async_fn_env#0}<risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>, risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>>>>, alloc::alloc::Global>>, (), madsim::sim::task::{impl#10}::spawn_inner::{closure_env#1}<tokio_metrics::task::Instrumented<futures_util::future::either::Either<await_tree::registry::{impl#1}::instrument::{async_fn_env#0}<risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>, risingwave_stream::task::stream_manager::{impl#2}::build_actors::{async_fn#0}::{async_block_env#2}>>>, alloc::sync::Weak<madsim::sim::task::TaskInfo>>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.4.0/src/raw.rs:563:17
  50: {closure#0}
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.22/src/sim/task/mod.rs:271:50
  51: do_call<madsim::sim::task::{impl#3}::run_all_ready::{closure_env#0}, bool>
             at /rustc/f0411ffcebcd7f75ac02ed45feb53ffd07b75398/library/std/src/panicking.rs:500:40
  52: try<bool, madsim::sim::task::{impl#3}::run_all_ready::{closure_env#0}>
             at /rustc/f0411ffcebcd7f75ac02ed45feb53ffd07b75398/library/std/src/panicking.rs:464:19
  53: catch_unwind<madsim::sim::task::{impl#3}::run_all_ready::{closure_env#0}, bool>
             at /rustc/f0411ffcebcd7f75ac02ed45feb53ffd07b75398/library/std/src/panic.rs:142:14
  54: run_all_ready
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.22/src/sim/task/mod.rs:271:17
  55: block_on<risingwave_simulation::main::{closure#0}::{async_block_env#0}>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.22/src/sim/task/mod.rs:238:13
  56: block_on<risingwave_simulation::main::{closure#0}::{async_block_env#0}>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.22/src/sim/runtime/mod.rs:126:9
  57: {closure#0}<risingwave_simulation::main::{closure#0}::{async_block_env#0}>
             at /Users/noelkwan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.22/src/sim/runtime/builder.rs:128:35
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### To Reproduce

1. Clone sqlsmith query snapshots. https://github.com/risingwavelabs/sqlsmith-query-snapshots/tree/main
2. Run the failed query set
```
RUST_BACKTRACE=1 MADSIM_TEST_SEED=22 RUST_LOG=info \
./target/sim/ci-sim/risingwave_simulation \
  --run-sqlsmith-queries ~/projects/sqlsmith-query-snapshots/failed/46
```

Sql: https://github.com/risingwavelabs/sqlsmith-query-snapshots/blob/main/failed/46/queries.sql

**Comment by zwang28** [comment]
minmal mv to reproduce (part of original m0):

```
create materialized view m as SELECT
            (992) AS col_0,
            (737) AS col_1,
            hop_3.date_time AS col_2
        FROM (tumble (alltypes1, alltypes1.c11, INTERVAL '64') AS tumble_2
            LEFT JOIN hop (auction, auction.date_time, INTERVAL '3600', INTERVAL '111600') AS hop_3 ON tumble_2.c11 = hop_3.expires);
```

**Comment by zwang28** [comment]
Explained in https://github.com/risingwavelabs/risingwave/issues/10495#issuecomment-1619370490

**Comment by zwang28** [comment]
Fixed by #10726
